tox.ini

```diff
@@ -69,6 +69,13 @@ allowlist_externals =
 commands =
     pytest tests/integration {posargs}
 
+[testenv:noextras]
+basepython = python3.11
+deps =
+  -r{toxinidir}/dev-noextras.requirements.txt
+commands =
+    coverage run -p -m pytest tests/unit/test_init_plugin_no_extras.py {posargs}
```

Comment:
Perhaps with appropriate use of conditional imports and `skip()`/`skipif()` etc, we should be able to run the full test suite without the extras installed. 

This would help to identify a critical type of error that hit us a few times in starlite earlier on, where the tests would always run with the full set of extras available. Some piece of non-optional functionality would be added, that inadvertently relies on having an extra available, and it isn't picked up by tests because the tests are only ever run with full extras available.

That was the reason for this: https://github.com/starlite-api/starlite/blob/624b0693c34b81f272b24fd385c2eb6bd77c5328/.github/workflows/publish.yaml#L6-L32, but I think with `tox` we can do it better as part of the usual test suite.